src/script/view_model/content/GiphyViewModel.ts

```diff
@@ -126,74 +125,81 @@ export class GiphyViewModel {
     }
   };
 
-  clickToSend = (): void => {
+  readonly clickToSend = (): void => {
     const selectedGif = this.selectedGif();
     if (selectedGif) {
       amplify.publish(WebAppEvents.EXTENSIONS.GIPHY.SEND, selectedGif.animated, this.query());
       this.selectedGif(undefined);
       this.modal.hide();
+      this.giphyRepository.resetOffset();
     }
   };
 
-  showGiphy = (query: string): void => {
+  readonly showGiphy = (query: string): void => {
     this.query(query);
     this.state(GiphyState.DEFAULT);
-    this._getRandomGif();
+    this.getGifs(true);
 
     if (!this.modal) {
       this.modal = new Modal('#giphy-modal', () => {
         this.modal = undefined;
+        this.giphyRepository.resetOffset();
       });
     }
 
     this.modal.show();
   };
 
-  _clearGifs = (): void => {
+  private readonly clearGifs = (): void => {
     this.gifs.removeAll();
     this.selectedGif(undefined);
     this.state(GiphyState.LOADING);
   };
 
-  _getRandomGif = async (): Promise<void> => {
+  private readonly getRandomGif = async (): Promise<void> => {
     const isStateError = this.state() === GiphyState.ERROR;
     if (isStateError) {
       return;
     }
 
-    this._clearGifs();
+    this.clearGifs();
 
     try {
       const gif = await this.giphyRepository.getRandomGif({tag: this.query()});
-      this.gif(gif);
-      this.gifs([this.gif()]);
-      this.selectedGif(this.gif());
+      this.currentGif(gif);
+      this.gifs([this.currentGif()]);
+      this.selectedGif(this.currentGif());
       this.state(GiphyState.RESULT);
     } catch (error) {
       this.logger.warn(error);
       this.state(GiphyState.ERROR);
     }
   };
 
-  _getRandomGifs = async (): Promise<void> => {
+  private readonly getGifs = async (displaySingleResult = false): Promise<void> => {
     const isStateError = this.state() === GiphyState.ERROR;
     if (isStateError) {
       return;
     }
 
-    this._clearGifs();
+    this.clearGifs();
 
     try {
       const gifs = await this.giphyRepository.getGifs(this.query());
       if (gifs.length === 0) {
         this.state(GiphyState.NO_SEARCH_RESULT);
         return;
       }
-      this.gifs(gifs);
-      if (gifs.length === 1) {
-        this.selectedGif(gifs[0]);
+
+      if (gifs.length === 1 || displaySingleResult === true) {
```

Comment:
I never liked comparing to === true.. is that needed here?

Reply:
Yes because IMO explicit > implicit :)